backend/src/main/java/com/redhat/cloud/notifications/routers/internal/TemplateResource.java

```diff
@@ -111,8 +112,8 @@ public InstantEmailTemplate createInstantEmailTemplate(@NotNull @Valid InstantEm
     @Path("/email/instant")
     @Produces(APPLICATION_JSON)
     @RolesAllowed(RBAC_INTERNAL_USER)
-    public List<InstantEmailTemplate> getAllInstantEmailTemplates() {
-        return templateRepository.findAllInstantEmailTemplates();
+    public List<InstantEmailTemplate> getAllInstantEmailTemplates(@QueryParam("application_id") @Valid UUID applicationId) {
```

Comment:
All our `@QueryParams` are currently camel cased in `backend`:
```suggestion
    public List<InstantEmailTemplate> getAllInstantEmailTemplates(@QueryParam("applicationId") UUID applicationId) {
```

Reply:
My suggestion above also removes `@Valid` because that annotation only works on a param that contains `javax.validation` constraints and `UUID` does not.